File: app/src/main/java/com/gaurav/newsapp/data/db/DataConverter.java
package com.gaurav.newsapp.data.db;

import androidx.room.TypeConverter;

import com.gaurav.newsapp.data.Article;
import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;

import java.lang.reflect.Type;
import java.util.List;

public class DataConverter {

    @TypeConverter
    public String fromArticleList(List<Article> articles) {
        if (articles == null) {
            return (null);
        }
        Gson gson = new Gson();
        Type type = new TypeToken<List<Article>>() {}.getType();
        String json = gson.toJson(articles, type);
        return json;
    }

    @TypeConverter
    public List<Article> toArticleList(String articleString) {
        if (articleString == null) {
            return (null);
        }
        Gson gson = new Gson();
        Type type = new TypeToken<List<Article>>() {}.getType();
        List<Article> countryLangList = gson.fromJson(articleString, type);
        return countryLangList;
    }
 }

File: app/src/main/java/com/gaurav/newsapp/data/db/DataConverter.kt
package com.gaurav.newsapp.data.db

import androidx.room.TypeConverter
import com.gaurav.newsapp.data.Article
import com.google.gson.Gson
import com.google.gson.reflect.TypeToken

class DataConverter {
    @TypeConverter
    fun fromArticleList(articles: List<Article?>?): String? {
        if (articles == null) {
            return null
        }
        val gson = Gson()
        val type =
            object : TypeToken<List<Article?>?>() {}.type
        return gson.toJson(articles, type)
    }

    @TypeConverter
    fun toArticleList(articleString: String?): List<Article>? {
        if (articleString == null) {
            return null
        }
        val gson = Gson()
        val type =
            object : TypeToken<List<Article?>?>() {}.type
        return gson.fromJson<List<Article>>(articleString, type)
    }
}